### ISSUE TITLE
can't serialize class org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema

### ISSUE THREAD

**Issue by antonstamov** [Stratio/spark-mongodb#65]
when I call in pySpark:
```
df = sql_context.jsonRDD(json_rdd)
    df.write.format("com.stratio.datasource.mongodb") \
        .mode("append").options(**options).save()
```
I get 
>  File "/usr/lib/spark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 330, in save
  File "/usr/lib/spark/python/lib/py4j-0.8.2.1-src.zip/py4j/java_gateway.py", line 538, in __call__
  File "/usr/lib/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 36, in deco
  File "/usr/lib/spark/python/lib/py4j-0.8.2.1-src.zip/py4j/protocol.py", line 300, in get_return_value
py4j.protocol.Py4JJavaError: An error occurred while calling o70.save.
: org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 5.0 failed 4 times, most recent failure: Lost task 1.3 in stage 5.0 (TID 12, ip-10-235-22-147.us-west-1.compute.internal): java.lang.IllegalArgumentException: can't serialize class org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
	at org.bson.BasicBSONEncoder._putObjectField(BasicBSONEncoder.java:299)
	at org.bson.BasicBSONEncoder.putIterable(BasicBSONEncoder.java:324)
	at org.bson.BasicBSONEncoder._putObjectField(BasicBSONEncoder.java:263)
	at org.bson.BasicBSONEncoder.putObject(BasicBSONEncoder.java:194)
	at org.bson.BasicBSONEncoder._putObjectField(BasicBSONEncoder.java:255)
	at org.bson.BasicBSONEncoder.putObject(BasicBSONEncoder.java:194)
	at org.bson.BasicBSONEncoder.putObject(BasicBSONEncoder.java:136)
	at com.mongodb.DefaultDBEncoder.writeObject(DefaultDBEncoder.java:36)
	at com.mongodb.BSONBinaryWriter.encodeDocument(BSONBinaryWriter.java:339)
	at com.mongodb.InsertCommandMessage.writeTheWrites(InsertCommandMessage.java:45)
	at com.mongodb.InsertCommandMessage.writeTheWrites(InsertCommandMessage.java:23)
	at com.mongodb.BaseWriteCommandMessage.encodeMessageBody(BaseWriteCommandMessage.java:69)
	at com.mongodb.BaseWriteCommandMessage.encodeMessageBody(BaseWriteCommandMessage.java:23)
	at com.mongodb.RequestMessage.encode(RequestMessage.java:66)
	at com.mongodb.BaseWriteCommandMessage.encode(BaseWriteCommandMessage.java:53)
	at com.mongodb.DBCollectionImpl.sendWriteCommandMessage(DBCollectionImpl.java:520)
	at com.mongodb.DBCollectionImpl.access$200(DBCollectionImpl.java:48)
	at com.mongodb.DBCollectionImpl$2.execute(DBCollectionImpl.java:470)
	at com.mongodb.DBCollectionImpl$2.execute(DBCollectionImpl.java:461)
	at com.mongodb.DBPort.doOperation(DBPort.java:187)
	at com.mongodb.DBTCPConnector.doOperation(DBTCPConnector.java:208)
	at com.mongodb.DBCollectionImpl.writeWithCommandProtocol(DBCollectionImpl.java:461)
	at com.mongodb.DBCollectionImpl.insertWithCommandProtocol(DBCollectionImpl.java:426)
	at com.mongodb.DBCollectionImpl.access$2000(DBCollectionImpl.java:48)
	at com.mongodb.DBCollectionImpl$Run$4.executeWriteCommandProtocol(DBCollectionImpl.java:849)
	at com.mongodb.DBCollectionImpl$Run$RunExecutor.execute(DBCollectionImpl.java:888)
	at com.mongodb.DBCollectionImpl$Run.executeInserts(DBCollectionImpl.java:862)
	at com.mongodb.DBCollectionImpl$Run.execute(DBCollectionImpl.java:738)
	at com.mongodb.DBCollectionImpl.executeBulkWriteOperation(DBCollectionImpl.java:149)
	at com.mongodb.DBCollection.executeBulkWriteOperation(DBCollection.java:1737)
	at com.mongodb.BulkWriteOperation.execute(BulkWriteOperation.java:109)
	at com.mongodb.casbah.BulkWriteOperation$$anonfun$2.apply(BulkWriteOperation.scala:88)
	at com.mongodb.casbah.BulkWriteOperation$$anonfun$2.apply(BulkWriteOperation.scala:88)
	at scala.util.Try$.apply(Try.scala:161)
	at com.mongodb.casbah.BulkWriteOperation.execute(BulkWriteOperation.scala:88)
	at com.stratio.datasource.mongodb.writer.MongodbBatchWriter$$anonfun$save$1.apply(MongodbBatchWriter.scala:44)
	at com.stratio.datasource.mongodb.writer.MongodbBatchWriter$$anonfun$save$1.apply(MongodbBatchWriter.scala:37)
	at scala.collection.Iterator$class.foreach(Iterator.scala:727)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1157)
	at com.stratio.datasource.mongodb.writer.MongodbBatchWriter.save(MongodbBatchWriter.scala:37)
	at com.stratio.datasource.mongodb.writer.MongodbWriter.saveWithPk(MongodbWriter.scala:82)
	at com.stratio.datasource.mongodb.MongodbDataFrame$$anonfun$saveToMongodb$1.apply(mongodbFunctions.scala:60)
	at com.stratio.datasource.mongodb.MongodbDataFrame$$anonfun$saveToMongodb$1.apply(mongodbFunctions.scala:56)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:902)
	at org.apache.spark.rdd.RDD$$anonfun$foreachPartition$1$$anonfun$apply$29.apply(RDD.scala:902)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:1850)
	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:1850)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:66)
	at org.apache.spark.scheduler.Task.run(Task.scala:88)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:214)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

Spark version 1.5.1 
runned from `spark-submit --packages com.stratio.datasource:spark-mongodb_2.10:0.10.1 test.py`

**Comment by marcovit79** [comment]
Have you some Array with element containing Struct ?
If so your problem could be related to mine (pull request 64).

That do not solve anything.
I wrote this comment only as a hint for the maintainer.

**Comment by antonstamov** [comment]
@marcovit79 yes, I have complex nested structure

**Comment by pmadrigal** [comment]
@antonstamov like @marcovit79 said, looks like  #64. It's already merged on branch-0.10, so you can test if it's fixed.

Thanks for your feedback!
